src/components/MatchInfo.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const searchQuery = styled.input`
+  border: red solid;
+`;
+class MatchInfo extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      searchQuery: '',
+      searchQueryValidate: false,
+      match: []
+    };
+  }
+
+  handleChange = event => {
+    this.setState({ searchQuery: event.target.value.trim() });
```

Comment:
`this.setState({ searchQuery: event.target.value.trim(), searchQueryValid: ---regex-check-here--- });`